src/xtgeo/grid3d/_gridprop_import_roff.py

```diff
@@ -1,18 +1,28 @@
 """Importing grid props from ROFF, binary"""
 
+from __future__ import annotations
 
 import warnings
+from typing import TYPE_CHECKING
 
 import numpy as np
 
 from xtgeo.common import null_logger
 
 from ._roff_parameter import RoffParameter
 
+if TYPE_CHECKING:
+    from xtgeo.common import _XTGeoFile
+    from xtgeo.grid3d import Grid, GridProperties, GridProperty
+
 logger = null_logger(__name__)
 
 
-def import_roff(pfile, name=None, grid=None):
+def import_roff(
+    pfile: _XTGeoFile,
+    name: str | None = None,
+    grid: Grid | GridProperty | GridProperties | None = None,
```

Comment:
@jcrivenaes / @eivindjahren Can one of you confirm the Grid union?

Reply:
I believe this should only be a Grid; the dict is unpacked into kwargs in the instantiation of a GridProperty (the _XTGeoFile) which can link back to a Grid instance